gwos-evm/polyjuice-tests/fuzz/gw-syscall-simulator/src/lib.rs

```diff
@@ -0,0 +1,714 @@
+    // check exists
+    match host.get_account_id_by_script_hash(&script_hash.into()) {
+        Ok(Some(_)) => return GW_ERROR_DUPLICATED_SCRIPT_HASH.into(),
+        Ok(None) => {}
+        Err(_err) => {
+            return ERROR;
+        }
+    }
+    // TODO: valide script
+    let id = match host.get_account_count() {
+        Ok(id) => id,
+        Err(_err) => {
+            return ERROR;
+        }
+    };
+
+    let account_nonce_key = build_account_field_key(id, GW_ACCOUNT_NONCE_TYPE);
+    host.state
+        .update_raw(account_nonce_key, H256::zero())
+        .expect("account nonce key");
+    host.run_result.write_data_hashes.insert(account_nonce_key);
+    let account_script_hash_key = build_account_field_key(id, GW_ACCOUNT_SCRIPT_HASH_TYPE);
+    host.state
+        .update_raw(account_script_hash_key, script_hash)
+        .expect("account script hash key");
+    host.run_result
+        .write_data_hashes
+        .insert(account_script_hash_key);
+    // script hash to id
+    let script_hash_to_id_value: H256 = {
+        let mut buf: [u8; 32] = H256::from_u32(id).into();
+        // the first 4 bytes is id, set exists flag(fifth byte) to 1
+        buf[4] = 1;
+        buf.into()
+    };
+    let script_hash_to_account_id_key = build_script_hash_to_account_id_key(&script_hash[..]);
+    host.state
+        .update_raw(script_hash_to_account_id_key, script_hash_to_id_value)
+        .expect("write script hash to account id key");
+    host.run_result
+        .write_data_hashes
+        .insert(script_hash_to_account_id_key);
+    // insert script
+    host.state.insert_script(script_hash, script);
+    host.state
+        .set_account_count(id + 1)
+        .expect("set account count");
+
+    let size_ptr = account_id_addr.as_mut().expect("casting pointer");
+    *size_ptr = id;
+
+    SUCCESS
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_load_tx(addr: *mut c_void, len: *mut u64) -> c_int {
+    if let Some(tx) = &HOST.lock().unwrap().tx {
+        store_data(addr, len, 0, tx);
+        SUCCESS
+    } else {
+        GW_ITEM_MISSING
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_load_block_info(addr: *mut c_void, len: *mut u64) -> c_int {
+    let data = HOST.lock().unwrap().block_info.as_slice().to_vec();
+    store_data(addr, len, 0, &data);
+    SUCCESS
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn gw_store_data(data_addr: *const u8, len: u64) -> c_int {
+    let data = load_bytes(data_addr, len);
+    let mut data_hash = [0u8; 32];
+    let mut hasher = new_blake2b();
+    hasher.update(data.as_ref());
+    hasher.finalize(&mut data_hash);
+    // insert data hash into SMT
+    let data_hash_key = build_data_hash_key(&data_hash);
+    let mut host = HOST.lock().unwrap();
+    host.state
+        .update_raw(data_hash_key, H256::one())
+        .expect("gw store data");
+    host.state.insert_data(data_hash.into(), data);
+
+    host.run_result.write_data_hashes.insert(data_hash_key);
```

Comment:
https://github.com/godwokenrises/godwoken/blob/develop/crates/generator/src/syscalls/mod.rs#L411

`data_hash_key` should not be stored into `write_data_hashes`